eclipse-collections-api/src/main/java/org/eclipse/collections/api/factory/map/ImmutableMapFactory.java

```diff
@@ -59,12 +60,37 @@
 
     <K, V> ImmutableMap<K, V> with(K key1, V value1, K key2, V value2, K key3, V value3, K key4, V value4);
 
+    /**
+     * @deprecated use {@link #withMap(Map)} instead (inlineable).
+     */
+    @Deprecated
```

Comment:
It's implementation has been deprecated ever since it was committed. I have just aligned the interface with its implementation.

https://github.com/eclipse/eclipse-collections/blob/417aa68093a26897ac46fa6ec7023b8ec0ea6ca3/eclipse-collections/src/main/java/org/eclipse/collections/impl/map/immutable/ImmutableMapFactoryImpl.java#L139-L146